openedx/core/djangoapps/catalog/utils.py

```diff
@@ -120,6 +122,10 @@ def get_programs(site=None, uuid=None, uuids=None, course=None):
         uuids = cache.get(SITE_PROGRAM_UUIDS_CACHE_KEY_TPL.format(domain=site.domain), [])
         if not uuids:
             logger.warning(u'Failed to get program UUIDs from the cache for site {}.'.format(site.domain))
+    elif organization:
+        uuids = cache.get(PROGRAMS_BY_ORGANIZATION_CACHE_KEY_TPL.format(org_key=organization))
```

Comment:
This could be DRYed up a bit as  `uuids = get_programs_for_organization(organization)`